src/MaxMind/Db/Reader.php

```diff
@@ -342,7 +342,11 @@ private function findMetadataStart(string $filename): int
                 break;
             }
 
-            $value = fread($handle, abs($markerLength));
+            /**
+             * METADATA_START_MARKER_LENGTH is a positive integer.
+             * @var int<0, max> $markerLength
+             */
```

Comment:
Sorry, I meant updating the type hint here:

https://github.com/maxmind/MaxMind-DB-Reader-php/blob/3c643c7de71165c0c2d6860bcc6129bd83ae2a66/src/MaxMind/Db/Reader.php#L29-L31

I think that should solve the issue and may result is slightly better checking generally.

Reply:
Done. I was afraid to touch that, in case `METADATA_START_MARKER_LENGTH` was used in other places. I did a quick text search and I see its only used here, so a type-hint at the definition would make more sense.